src/main/java/com/refinedmods/refinedstorage/apiimpl/network/grid/CraftingGridBehavior.java

```diff
@@ -71,25 +84,54 @@ public void onCrafted(INetworkAwareGrid grid, ICraftingRecipe recipe, PlayerEnti
 
     @Override
     public void onCraftedShift(INetworkAwareGrid grid, PlayerEntity player) {
+        CraftingInventory matrix = grid.getCraftingMatrix();
+        INetwork network = grid.getNetwork();
         List<ItemStack> craftedItemsList = new ArrayList<>();
-        int amountCrafted = 0;
         ItemStack crafted = grid.getCraftingResult().getStackInSlot(0);
 
+        int amountCrafted = 0;
+        boolean useNetwork = network != null;
+
+        IStackList<ItemStack> networkCraftingItems = null;
+        if (useNetwork) {
+            // We need a modifiable list of the items in storage that are relevant for this craft.
+            networkCraftingItems = API.instance().createItemStackList();
+            for (int i = 0; i < matrix.getSizeInventory(); ++i) {
```

Comment:
Why don't we use a copy of the network item stack list straight away? Instead of making this filtered list.

Reply:
This is faster. Copying all thousands of items with their NBT is not necessary here. We just need the up to 9 items that correspond to the items in the crafting grid. 

I'm later also calling `IStackList.remove()` which needs to iterate over the entire list.

Reply:
Yeah, forgot NBT copying (and stack copying in general) is super slow. Can you extract this logic to another method, then? Makes the code more readable.